projects/step-frontend/src/lib/modules/execution/components/execution-tabs/execution-tabs.component.ts

```diff
@@ -1,18 +1,72 @@
 import { Component, EventEmitter, Input, Output } from '@angular/core';
-import { ExecutionTab } from '../../shared/execution-tab';
 import { downgradeComponent, getAngularJSGlobal } from '@angular/upgrade/static';
-import { AJS_MODULE } from '@exense/step-core';
+import { AJS_MODULE, Dashlet, Tab, ViewRegistryService } from '@exense/step-core';
+import {
+  ExecutionTabsCommand,
+  ExecutionsTabsControlService,
+  ExecutionTabsCommandType,
+} from '../../services/executions-tabs-control.service';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'step-execution-tabs',
   templateUrl: './execution-tabs.component.html',
   styleUrls: ['./execution-tabs.component.scss'],
 })
 export class ExecutionTabsComponent {
-  @Input() tabs: ExecutionTab[] = [];
+  @Input() executionId?: string;
   @Input() activeTabId?: string;
   @Output() activeTabIdChange = new EventEmitter<string>();
-  @Output() closeTab = new EventEmitter<string>();
+
+  readonly tabs: Array<Tab>;
+
+  private subscription: Subscription;
+
+  constructor(
+    private _viewRegistryService: ViewRegistryService,
+    private _executionTabsControlService: ExecutionsTabsControlService
+  ) {
+    this.tabs = this._viewRegistryService
+      .getDashlets('executionTab')
+      .filter(function (dashlet) {
+        return dashlet.isEnabledFct ? dashlet.isEnabledFct() : false;
+      })
+      .map<Tab>((dashlet) => this.dashletToTab(dashlet));
+    this.subscription = this._executionTabsControlService.controlSubject$.subscribe((command: ExecutionTabsCommand) => {
+      this.handleExecutionTabsCommand(command);
+    });
+  }
+
+  private handleExecutionTabsCommand(command: ExecutionTabsCommand) {
+    if (command.executionId === this.executionId) {
+      switch (command.command) {
+        case ExecutionTabsCommandType.DISABLE:
+          this.disableTabById(command.id);
+          break;
+        default:
+          break;
+      }
+    }
+  }
+
+  private disableTabById(id: string): void {
+    let tab = this.findTabById(id);
```

Comment:
The idea here is that in the future there will be more commands, like `enableTabById`. So this method would be reused. But I tweaked it the way you suggested using find instead of filter:
```
  private findTabById(id: string): Tab | undefined {
    return this.tabs.find((tab) => tab.id === id);
  }
```